helios-client/src/main/java/com/spotify/helios/common/descriptors/HealthCheck.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.common.descriptors;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.Lists;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.lang.String.format;
+
+/**
+ * Defines the health check for the Helios job. There are 3 types of health checks.
+ *
+ * HTTP healthcheck:
+ *
+ * <pre>
+ * {
+ *   "type" : "http",
+ *   "path" : "/path/to/healthcheck",
+ *   "port" : "http-admin"
+ * }
+ * </pre>
+ *
+ * TCP healthcheck:
+ *
+ * <pre>
+ * {
+ *   "type" : "tcp",
+ *   "port" : "http-admin"
+ * }
+ * </pre>
+ *
+ * Command-based healthcheck:
+ *
+ * <pre>
+ * {
+ *   "type" : "cmd",
+ *   "cmd" : "bash -c '/usr/bin/curl 127.0.0.1:9200/_cluster/health?pretty=true | grep green'"
+ * }
+ * </pre>
+ */
+
+public class HealthCheck extends Descriptor {
+
+  public static final String HTTP = "http";
+  public static final String TCP = "tcp";
+  public static final String CMD = "cmd";
+  public static final List<String> VALID_HEALTH_CHECK_TYPES = Lists.newArrayList(HTTP, TCP, CMD);
+
+  private final String type;
+  private final String path;
+  private final String port;
+  private final String cmd;
```

Comment:
Another option would be to make each HealthCheck it's own object. This has a few benefits.
1. More type safe than a map of Objects, which can be messy when getting values out of the map because you need to cast everything. Also not sure if a map of Objects will cause issues when deserializing.
2. Users can write very concise code when creating healthchecks in TempJobs, e.g. HeathCheck.newHttpCheck("/healthcheck", 8080);
3. More efficient wrt serialization than a single object, because you will serialize only the fields needed for that health check type.